internal/buffer/buffer.go

```diff
@@ -950,6 +952,8 @@ func (b *Buffer) UpdateRules() {
 					break
 				}
 			}
+			// TODO: fix this mess. For each include, we should include
+			// exactly one file: either a built-in one or a user's custom one
```

Comment:
So, I was wrong about this one too: since #3066 we already ensure that exactly one file will be included.

This `len(files) >= len(includes)` check is now not strictly necessary, but it provides a simple optimization (to exit the loop earlier), so we can keep it.